### PR TITLE
doc.protocol: Add gRPC documentation

### PR DESCRIPTION
**Pull request by romainletendart**
The main intent of this PR is to discuss about error handling with the new unified gRPC interface.
This also makes a good opportunity to start documenting (or at least listing) all the gRPC interfaces available.

### CODE REVIEW

File: doc/protocol/grpc.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+It is easy to miss for someone who would want to implement `Get` or `Set` and could lead to hidden errors.
+
+`Subscribe` and `GetServerInfo` don't have such error fields in their respective `*Response` protobuf messages.
+Instead, errors if any will result in an exception being raised (e.g. `grpc.aio.AioRpcError` in python's case) and that
+already comes without further efforts with protoc-generated code.
+
+In some cases though, `Get` and `Set` might still raise `grpc.aio.AioRpcError` e.g. because of network errors.
+This means the client needs to handle both `grpc.aio.AioRpcError` errors and response-message-based errors.
+
+What response-message-based errors do provide however are much more detailed errors which is beneficial for the user.
```
Comment (erikbosch) [inline doc/protocol/grpc.md:29]: Maybe rephrase this sentence, took some time for me to understand

**Review by erikbosch — APPROVED** [review]

**Comment by SebastianSchildt** [comment]
@romainletendart  @argerus Not sure, what the status is here currently, the text still lists "current state" and "desired state" for error handling. Wasn't the plan to move over to "desired" implementation wise, and just document that here?

**Comment by romainletendart** [comment]
@SebastianSchildt I would not document something as "implemented" if the implementation is not there yet.
I would instead mark this PR as blocked by https://github.com/eclipse/kuksa.val/issues/385.
Once we have implemented the new error handling, we update this PR accordingly I would say.

**Comment by lukasmittag** [comment]
Does this need a merge or further work?

**Comment by erikbosch** [comment]
@SebastianSchildt - I think we need to have a general discussion/decision on how we want error handling to look like in the long term, and as well what changes we want to do in the short time.

For now I think the discussion and work-in-progress fits better in a Wiki-page so i created https://github.com/eclipse/kuksa.val/wiki/KUKSA.val-gRPC-interfaces. I suggest this PR can be closed, when we have agreed on the way forward we can create a new PR
